HIP-6.md

```diff
@@ -0,0 +1,147 @@
+| hip | title                   | status | author | created    |
+| --- | ----------------------- | ------ | ------ | ---------- |
+| 6   | Validator announcements | Draft  | asaj   | 2022-12-21 |
+
+### **Brief Summary / Abstract**
+
+Defines an optional specification for Hyperlane validators to announce themselves and the location of their signed checkpoints.
+
+### **Motivation**
+
+Hyperlane relayers need to aggregate validator signed checkpoints in order to deliver interchain messages.
+
+At present, relayers are manually configured with a list of validators and the location of their signed checkpoints.
+
+Relying on manual configuration creates friction to the introduction of additional validators via sovereign consensus.
+
+This HIP defines a standard for validators to announce themselves to relayers via a smart contract on the underlying blockchain.
+
+Relayers should query this contract to discover the location of validator signed checkpoints.
+
+### **Tech Spec**
+
+#### Announcment attestations
```

Comment:
Do these validator announcements only apply to validators that plan to sign checkpoints in the manner outlined by #6?
Wondering if we should add a "signatureType" or something to allow for other types of attestations in the future (e.g. different signature schemes or proofs or something).
Also wondering if we should explicitly outline the expected content of the S3 buckets provided here somewhere- e.g. document `latest_checkpoint.json` and how each signed checkpoint should be provided

Reply:
> Do these validator announcements only apply to validators that plan to sign checkpoints in the manner outlined by https://github.com/hyperlane-xyz/hips/pull/6?

I hadn't really thought about other uses. What were you imagining?

> Also wondering if we should explicitly outline the expected content of the S3 buckets provided here somewhere- e.g. document latest_checkpoint.json and how each signed checkpoint should be provided

Agreed that this could use specification. Maybe in its own AWS S3 storage HIP? Or alternatively in #6 ?

Reply:
> I hadn't really thought about other uses. What were you imagining?

any protocol could tap into staked hyperlane validators for security on arbitrary data maybe?

Reply:
>Agreed that this could use specification. Maybe in its own AWS S3 storage HIP? Or alternatively in https://github.com/hyperlane-xyz/hips/pull/6 ?

IMO we can spec this API and schema here, independently of storage provider

Reply:
@yorhodes we *can*, and probably should at some point, but I'm not sure writing up a spec for what's already implemented would be as useful as some of the other things we could be working on

Reply:
Agree that specifying this is useful but ultimately a nice to have